lib/Sema/TypeCheckStmt.cpp

```diff
@@ -765,10 +765,22 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
 
       // If someone is using an unlabeled break inside of an 'if' or 'do'
       // statement, produce a more specific error.
-      if (S->getTargetName().empty() && !ActiveLabeledStmts.empty() &&
-          (isa<IfStmt>(ActiveLabeledStmts.back()) ||
-           isa<DoStmt>(ActiveLabeledStmts.back())))
-        diagid = diag::unlabeled_break_outside_loop;
+      if (S->getTargetName().empty() && !ActiveLabeledStmts.empty()) {
+        if (isa<IfStmt>(ActiveLabeledStmts.back()) ||
+            isa<DoStmt>(ActiveLabeledStmts.back())) {
+          diagid = diag::unlabeled_break_outside_loop;
+        }
+        
+       
+        if (isa<GuardStmt>(ActiveLabeledStmts.back())) {
+          if (ActiveLabeledStmts.size() >= 2 &&
+              std::any_of(ActiveLabeledStmts.rbegin(), ActiveLabeledStmts.rend(),
+                          [&](Stmt *S) -> bool { return isa<IfStmt>(S) || isa<DoStmt>(S); })) {
+
+            diagid = diag::unlabeled_break_outside_loop;
+          }
+        }
+      }
```

Comment:
Oh! duh, (to myself) what were you thinking. Thanks for pointing out. I've stolen your snippet and copied-pasted and pushed. :)